src/routes/hardwareRouter.ts

```diff
@@ -1,11 +1,33 @@
 import { Router } from "express";
 import { HardwareController } from "../controllers/hardwareController";
 import { checkIsLoggedIn, checkIsVolunteer, checkIsOrganizer } from "../util/user";
+import { HardwareService } from "../services/hardware/hardwareService";
+import { HardwareItem, ReservedHardwareItem, User, Team } from "../db/entity/hub";
+import { getConnection } from "typeorm";
+import { ReservedHardwareService } from "../services/hardware";
+import { TeamService } from "../services/teams/teamService";
+import { UserService } from "../services/users";
 
 export const hardwareRouter = (): Router => {
-  const router = Router();
+  const reservedHardwareService: ReservedHardwareService = new ReservedHardwareService(
```

Comment:
I just realized that instantiating services like this is a bit of an anti-pattern in terms of testability because we can't mock the created services. Ideally the services should be passed in the constructor or better yet we could implement dependency injection to be used throughout the entire codebase (would probably clean up the services as well) but I do realize that doing this would involve quite a bit of work and I'm not sure it's entirely worth it. 

But yeah I'm not too bothered either way but DI may be a good idea for the future. [This](http://inversify.io/) might be a good choice. It's very popular and is also built on TS so it should be reliable.

Reply:
I had a look into inversify and it looks like a really nice addition. But it would be quite a lot of work, but I think it is definitely doable with the current project architecture. I have referenced this comment in a new issue

Reply:
Cool